### PR TITLE
Update partners page to Bootstrap 4

### PR DESCRIPTION
**Pull request by jsnshrmn**
[//]: # (Thank you for uploading a PR to the Wikipedia Library!)

## Description
Updates `/partners/` page to reuse as much of the bootstrap 4 `/users/my_library/` code as possible. The content and filtering is still very similar to the existing `/partners/` page, with live text filtering being the only functional addition. This does also include fixes for a few unrelated but minor annoyances that came up: tests were not passing locally with the django debug toolbar enabled, and alert messages were displayed twice.

## Rationale
Bootstrap 3 is EOL

## Phabricator Ticket
https://phabricator.wikimedia.org/T293986
addresses part of
https://phabricator.wikimedia.org/T291487
by omitting the 'apply to multiple partners' button in the new template.

## How Has This Been Tested?
Since the existing textual content and filtering options are mostly unchanged, existing tests all pass. We had a review of the visual design, in which we identified and resolved numerous visual inconsistencies.

## Screenshots of your changes (if appropriate):
image

## Types of changes
What types of changes does your code introduce? Add an `x` in all the boxes that apply:
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Minor change (fix a typo, add a translation tag, add section to README, etc.)

### CODE REVIEW

File: TWLight/message_storage.py [hunk 1]
```diff
@@ -0,0 +1,12 @@
+from django.contrib.messages.storage.session import SessionStorage
+from django.contrib.messages.storage.base import Message
+from .view_mixins import DedupMessageMixin
+
+
+class SessionDedupStorage(DedupMessageMixin, SessionStorage):
```
Comment (suecarmol) [inline TWLight/message_storage.py:6]: Super interesting solution!

File: TWLight/resources/templates/resources/partner_list.html [hunk 2]
```diff
@@ -51,3 +40,84 @@ <h2>{% trans "Browse Partners" %}</h2>
...
+        displayPartnerTile(collectionElems, filteredCollections, pattern);
+      }
+      else if (searchResults.length === 0 && pattern === ""){
+        // No search text and no search results, display everything again
+        displayPartnerTile(collectionElems, filteredCollections, pattern);
+      }
+      else{
+        // Show search results
+        var favExpiredCount = 0;
+        var userExpiredCount = 0;
```
Comment (suecarmol) [inline TWLight/resources/templates/resources/partner_list.html:91]: I don't think these variables are used here

**Review by suecarmol — COMMENTED** [review]
This is looking great! I just have a very minor change request before we can merge this

File: TWLight/resources/templates/resources/partner_list.html (hunk 2 above)
Comment (jsnshrmn) [reply TWLight/resources/templates/resources/partner_list.html]: oopsie!

Comment (jsnshrmn) [reply TWLight/resources/templates/resources/partner_list.html]: done!

**Review by suecarmol — APPROVED** [review]

**Pull request merged**